package.json

```diff
@@ -26,7 +26,7 @@
   "scripts": {
     "prerun": "npm run-script build",
     "run": "jpm run --addon-dir dist/ ${npm_config_jpm_runflags} --prefs jpm-prefs.json",
-    "test": "echo \"Error: no test specified\" && exit 1",
+    "test": "mocha-webpack --opts ./mocha-webpack.opts --webpack-config ./webpack.config.test.js -r test/setup.js test",
```

Comment:
It's generally recommended to have a "pretest" script that ensures linting is run:

```
  "pretest": "npm run lint"
```

Reply:
I'd actually prefer to go the other direction and turn off auto-linting for the `build` command too. It's pretty frustrating when I'm just slapping code together for debugging and I have to fix linter issues before I can actually test my code. If CI is always linting, we should never run into lint failures accumulating...

As a compromise, I guess we could add a `NO_LINT` environment variable, but I'm a little worried that that would make the resulting commands POSIX-only (I sort of doubt that the Windows shell syntax is identical).

Reply:
[`cross-env`](https://github.com/kentcdodds/cross-env#readme) takes care of differences between POSIX and Windows.

I understand not wanting the lint to get in the way, but my experience has been it has fixed more problems than it causes -- even in troubleshooting mode.  Especially with `no-console` set to "warn" as it is here.

Reply:
My main issue is that the linter tries to address two classes of problems: stuff that would get caught by `-Wall` and true lint issues (e.g. requiring a space after a comment token). The latter problems are usually just noise that interrupt me during debugging, whereas the former are actually useful.

Maybe running lint as a pre-commit hook would be a good compromise? Then every commit should be fully-linted.

Reply:
Hooks in github would be better, although I'd still want linting run as part of Travis (so maybe I don't trust all the hooks (-: ).

The current setup is ok, and I'd rather have this than wait for "perfect".